pkg/content_download.go

```diff
@@ -0,0 +1,84 @@
+package pkg
+
+import (
+	"github.com/wttech/aemc/pkg/common/filex"
+	"github.com/wttech/aemc/pkg/common/pathx"
+	"github.com/wttech/aemc/pkg/content"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	FilterXml = "filter.xml"
+)
+
+type Downloader struct {
+	config *content.Opts
+}
+
+func NewDownloader(config *content.Opts) *Downloader {
+	return &Downloader{
+		config: config,
+	}
+}
+
+func (c Downloader) DownloadPackage(packageManager *PackageManager, pid string, roots []string, filter string) (string, error) {
```

Comment:
I am seeing it more like 

`PackageManager.DownloadPaths(localFile, pid, rootPaths, filter)`

as it has so close relationship with pkg manager; maybe it just needs to be a part of pkg mgr?:) I am open for it